torchgeo/trainers/change.py

```diff
@@ -235,7 +313,18 @@ def _shared_step(self, batch: Any, batch_idx: int, stage: str) -> Tensor:
                     keepdim=True,
                 )
                 batch = aug(batch)
-                batch['prediction'] = (y_hat.sigmoid() >= 0.5).long()
+                match self.hparams['task']:
+                    case 'binary':
+                        prediction = (y_hat.sigmoid() >= 0.5).long()
+                        # Restore channel dimension for plotting compatibility
+                        batch['prediction'] = prediction.unsqueeze(1)
+                    case 'multilabel':
+                        batch['prediction'] = (y_hat.sigmoid() >= 0.5).long()
```

Comment:
This is the only line missing coverage, and will be hard to cover since we don't yet have any real multilabel datasets. Let's combine `'binary' | 'multilabel'` and then have a separate if-statement later for binary plot squeezing.